contracts/contracts/trees/AccQueue.sol

```diff
@@ -330,12 +331,14 @@ abstract contract AccQueue is Ownable, Hasher {
             );
 
             // Increment the next subroot counter
-            nextSubRootIndex ++;
+            _nextSubRootIndex ++;
 
             // Increment the ops counter
             queueOpsPerformed ++;
         }
 
+        _nextSubRootIndex = nextSubRootIndex;
```

Comment:
This should instead be `nextSubRootIndex = _nextSubRootIndex` since `_nextSubRootIndex` is the temporary value that is holding the final result from the for loop

Reply:
This was fixed in a later commit but then reverted to the original implementation as it was causing some errors in the tests. Will try and find a solution but for now the state update will be within the loop.

Reply:
As this wasn't a vulnerability but instead a gas optimization, it's okay to merge it in this PR and open a separate issue documenting that this can potentially be a way to improve gas efficiency for this function.